lang/cs/Org.Apache.REEF.Client/YARN/LegacyJobResourceUploader.cs

```diff
@@ -0,0 +1,100 @@
+    /// Uploading of DFS is done via calling into org.apache.reef.bridge.client.JobResourceUploader
+    /// Uploading via Java is necessary to support Hadoop versions before v2.7.0 because
+    /// we need to find modification time with accuracy of seconds to use archive as resource for YARNRM
+    /// <see cref="Org.Apache.REEF.IO.FileSystem.Hadoop.HadoopFileSystem"/> implementation uses HDFS
+    /// commandline shell which does not provide modification time with accuracy of seconds until v2.7.1
+    /// </summary>
+    internal sealed class LegacyJobResourceUploader : IJobResourceUploader
+    {
+        private static readonly Logger Log = Logger.GetLogger(typeof(LegacyJobResourceUploader));
+
+        private static readonly string JavaClassNameForResourceUploader =
+            @"org.apache.reef.bridge.client.JobResourceUploader";
+
+        private readonly IJavaClientLauncher _javaLauncher;
+        private readonly IJobSubmissionDirectoryProvider _jobSubmissionDirectoryProvider;
+        private readonly IResourceArchiveFileGenerator _resourceArchiveFileGenerator;
+        private readonly IFile _file;
+
+        [Inject]
+        private LegacyJobResourceUploader(IJavaClientLauncher javaLauncher,
+            IJobSubmissionDirectoryProvider jobSubmissionDirectoryProvider,
+            IResourceArchiveFileGenerator resourceArchiveFileGenerator,
+            IFile file)
+        {
+            _file = file;
+            _resourceArchiveFileGenerator = resourceArchiveFileGenerator;
+            _jobSubmissionDirectoryProvider = jobSubmissionDirectoryProvider;
+            _javaLauncher = javaLauncher;
+        }
+
+        public JobResouce UploadJobResouce(string driverLocalFolderPath)
+        {
+            driverLocalFolderPath = driverLocalFolderPath.TrimEnd('\\') + @"\";
+            string driverUploadPath = _jobSubmissionDirectoryProvider.GetJobSubmissionDirectory().TrimEnd('/') + @"/";
+            Log.Log(Level.Info, "DriverFolderPath: {0} DriverUploadPath: {1}", driverLocalFolderPath, driverUploadPath);
+
+            var archivePath = _resourceArchiveFileGenerator.CreateArchiveToUpload(driverLocalFolderPath);
+
+            var resourceDetailsOutputPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N"));
+            _javaLauncher.Launch(JavaClassNameForResourceUploader,
+                archivePath,
+                driverUploadPath,
+                resourceDetailsOutputPath);
+
+            return ParseGeneratedOutputFile(resourceDetailsOutputPath);
+        }
+
+        private JobResouce ParseGeneratedOutputFile(string resourceDetailsOutputPath)
+        {
+            if (!_file.Exists(resourceDetailsOutputPath))
+            {
+                Exceptions.Throw(
+                    new FileNotFoundException("Could not find resource details file " + resourceDetailsOutputPath),
+                    Log);
+            }
+
+            string fileContent = _file.ReadAllText(resourceDetailsOutputPath).Trim();
```

Comment:
Chiefly defensive programming as the text is coming from a different environment (Java)